functions/Set-DbaPrivilege.ps1

```diff
@@ -92,45 +92,80 @@ function Convert-UserNameToSID ([string] `$Acc ) {
+                                        }
                                     }
                                     else {
-                                        Write-Warning "$acc already has Batch Logon Privilege on $env:ComputerName"
+                                        #here-strings require the tabbing to get all messed up.
+                                        $NewContent = @"
+[Privilege Rights]
+SeBatchLoginRight = *$SID
+"@
+                                        (Get-Content $tempfile) -replace "[Privilege Rights]", $NewContent | Set-Content $tempfile
+                                        Write-Verbose "Added new line for Batch Logon Privilege Privileges and granted $acc the privilege on $env:ComputerName" 
                                     }
                                 }
                             }
                             if ('IFI' -in $Type) {
-                                $IFIline = Get-Content $tempfile | Where-Object { $_ -match "SeManageVolumePrivilege" }
                                 ForEach ($acc in $SQLServiceAccounts) {
                                     $SID = Convert-UserNameToSID -Acc $acc;
-                                    if ($IFIline -notmatch $SID) {
-                                        (Get-Content $tempfile) -replace "SeManageVolumePrivilege = ", "SeManageVolumePrivilege = *$SID," |
-                                        Set-Content $tempfile
-                                        Write-Verbose "Added $acc to Instant File Initialization Privileges on $env:ComputerName"
-                                    }
+                                    $IFIline = Get-Content $tempfile | Where-Object { $_ -match "SeManageVolumePrivilege" }
+                                    if ($IFILine) {
+                                        if ($IFIline -notmatch $SID) {
+                                            (Get-Content $tempfile) -replace "SeManageVolumePrivilege = ", "SeManageVolumePrivilege = *$SID," |
+                                            Set-Content $tempfile
+                                            Write-Verbose "Added $acc to Instant File Initialization Privileges on $env:ComputerName"
+                                        }
+                                        else {
+                                            Write-Warning "$acc already has Instant File Initialization Privilege on $env:ComputerName"
+                                        }
+                                    }    
                                     else {
-                                        Write-Warning "$acc already has Instant File Initialization Privilege on $env:ComputerName"
+                                    
+                                        #here-strings require the tabbing to get all messed up.
+                                        $NewContent = @"
+[Privilege Rights]
+SeManageVolumePrivilege = *$SID
+"@
```

Comment:
We can't stand here strings....you can just put this as a regular string so it formats properly

Reply:
I'm using the here-string for the carriage return between the section title in the file and the entries.  I don't like them either, but seemed the simplest way to to do this.

Reply:
You can use "`n" for cariage return in a normal string.

So this would serve the same purpose `"[Privilege Rights]`nSeManageVolumePrivilege = *$SID"